utils/global-state-update-gen/src/utils/account_manager.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::BTreeMap;
+
+use casper_engine_test_support::LmdbWasmTestBuilder;
+use casper_types::{
+    account::Account, AccessRights, CLValue, Key, PublicKey, StoredValue, URef, U512,
+};
+
+use rand::prelude::*;
+
+use crate::utils::print_entry;
+
+pub struct AccountManager {
+    accounts: BTreeMap<PublicKey, Account>,
+}
+
+impl AccountManager {
+    /// Creates a new instance of `AccountManager`.
+    pub fn new() -> Self {
+        Self {
+            accounts: Default::default(),
+        }
+    }
+
+    /// Gets the account for the given public key.
+    /// If the account doesn't exist, it prints the entries necessary to create it and returns the
+    /// created account.
+    pub fn get_or_create_account(
+        &mut self,
+        builder: &mut LmdbWasmTestBuilder,
+        public_key: &PublicKey,
+    ) -> &Account {
+        // unfortunately, `if let Some(account) = self.accounts.get(&public_key)` upsets the borrow
```

Comment:
```suggestion
        match self.accounts.entry(public_key.clone()) {
            Entry::Vacant(vac) => {
                let account_hash = public_key.to_account_hash();
                let account = match builder.get_account(account_hash) {
                    Some(account) => account,
                    None => Self::create_account(public_key.clone()),
                };
                vac.insert(account)
            }
            Entry::Occupied(occupied) => occupied.into_mut(),
        }
```

And `fn create_account` doesn't need a `&self` anymore

Reply:
Nice one!